tracer/src/Datadog.Trace/Agent/Transports/HttpClientRequest.cs

```diff
@@ -76,6 +77,41 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new HttpClientResponse(response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            using var formDataContent = new MultipartFormDataContent();
+            _request.Content = formDataContent;
+
+            foreach (var item in items)
+            {
+                // Adds a form data item
+                if (item.ContentInBytes is { } arraySegment)
+                {
+                    var content = new ByteArrayContent(arraySegment.Array, arraySegment.Offset, arraySegment.Count);
+                    content.Headers.ContentType = new MediaTypeHeaderValue(item.ContentType);
+                    formDataContent.Add(content, item.Name, item.FileName);
+                    Log.Debug("Adding to Multipart Byte Array | Name: {Name} | FileName: {FileName} | ContentType: {ContentType}", item.Name, item.FileName, item.ContentType);
```

Comment:
nit: I would probably add the log message first in case something here throws
```suggestion
                    Log.Debug("Adding to Multipart Byte Array | Name: {Name} | FileName: {FileName} | ContentType: {ContentType}", item.Name, item.FileName, item.ContentType);
                    content.Headers.ContentType = new MediaTypeHeaderValue(item.ContentType);
                    formDataContent.Add(content, item.Name, item.FileName);
```